pkg/providers/imagefamily/nodeimage_test.go

```diff
@@ -171,39 +200,101 @@ var _ = Describe("NodeImageProvider tests", func() {
 			ctx = options.ToContext(ctx, testOptions)
 		})
 
-		It("should match expected images for Ubuntu2204", func() {
+		It("should match expected images for default Ubuntu2204 when FIPSMode is explicitly set to Disabled", func() {
+			nodeClass.Spec.ImageFamily = lo.ToPtr(v1beta1.Ubuntu2204ImageFamily)
+			nodeClass.Spec.FIPSMode = lo.ToPtr("Disabled")
+
 			foundImages, err := nodeImageProvider.List(ctx, nodeClass)
 			Expect(err).ToNot(HaveOccurred())
 			Expect(foundImages).To(ContainElements(getExpectedTestSIGImages(*nodeClass.Spec.ImageFamily, sigImageVersion, kubernetesVersion)))
 		})
 
+		It("should match expected images for default Ubuntu2204 when FIPSMode is not explicitly set", func() {
+			nodeClass.Spec.ImageFamily = lo.ToPtr(v1beta1.Ubuntu2204ImageFamily)
+
+			foundImages, err := nodeImageProvider.List(ctx, nodeClass)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(foundImages).To(ContainElements(getExpectedTestSIGImages(*nodeClass.Spec.ImageFamily, sigImageVersion, kubernetesVersion)))
+		})
+
+		//TODO: Modify when Ubuntu 22.04 with FIPS becomes available
+		It("should match expected images for FIPS Ubuntu2204 when FIPSMode is explicitly set to FIPS", func() {
+			nodeClass.Spec.ImageFamily = lo.ToPtr(v1beta1.Ubuntu2204ImageFamily)
+			nodeClass.Spec.FIPSMode = lo.ToPtr("FIPS")
```

Comment:
I think it might be worth adding a new `Context` in here for the FIPS tests, at least for the ones where FIPS is enabled.

If you do it just for the enabled cases you could do a `BeforeEach` which contains the `nodeClass.Spec.FIPSMode = lo.ToPtr("FIPS")`

If you do it more generically, I still think it could be nice to grouping of like tests, since its testing the sub cases of SIG images surrounding FIPS.